pkg/manager/agent.go

```diff
@@ -111,6 +111,15 @@ func (a *agentImpl) StartAgent(ctx context.Context, opts *StartAgentOptions) err
 					os.Setenv(key, val)
 				}
 
+				if opts.Manager.Provider == ProviderGithubCRI {
+					for key, val := range *allocationDetails.GhRunnerApplicationDetails.Variables {
```

Comment:
Add null checks for this dereference if not present

Reply:
Added above this.